### PR TITLE
Merge audit v2 remediation to main

### PR DESCRIPTION
**Pull request by yorhodes**
### Description

merge [`audit-v2`](https://github.com/hyperlane-xyz/hyperlane-monorepo/releases/tag/audit-v2) => `main`

### CODE REVIEW

File: .github/workflows/node.yml [hunk 1]
```diff
@@ -3,9 +3,9 @@ name: node
 on:
   # Triggers the workflow on push or pull request events but only for the main and v2 branches
   push:
-    branches: [main]
+    branches: [main, audit-v2-remediation]
```
Comment (asaj) [inline .github/workflows/node.yml:6]: I don't *think* this is necessary to merge to main? Ditto for below

File: solidity/contracts/OwnableMulticall.sol [hunk 2]
```diff
@@ -32,23 +33,28 @@ contract OwnableMulticall is OwnableUpgradeable {
...
+        resolveCalls = new Result[](length);
+        for (uint256 i = 0; i < length; i++) {
+            (bool success, bytes memory returnData) = calls[i].to.staticcall(
                 calls[i].data
             );
-            require(success, "Multicall: call failed");
-            resolveCalls[i] = bytes.concat(callbacks[i], returnData);
+            resolveCalls[i] = Result(
+                success,
+                bytes.concat(callbacks[i], returnData)
```
Comment (asaj) [inline solidity/contracts/OwnableMulticall.sol:49]: Feels kind of weird to concat if success == False, no?

Comment (asaj) [reply solidity/contracts/OwnableMulticall.sol:49]: I guess we don't do anything with that if success==false, in which case maybe it's an opportunity for gas savings?

File: solidity/contracts/OwnableMulticall.sol [hunk 3]
```diff
@@ -57,4 +63,19 @@ contract OwnableMulticall is OwnableUpgradeable {
...
         }
     }
+
+    function resolveResults(address to, Result[] memory results) internal {
+        for (uint256 i = 0; i < results.length; i += 1) {
+            if (results[i].success) {
+                (bool success, bytes memory returnData) = to.call(
+                    results[i].data
+                );
+                if (!success) {
```
Comment (asaj) [inline solidity/contracts/OwnableMulticall.sol:73]: `require(success, returnData)` doesn't work?

File: solidity/contracts/isms/MultisigIsm.sol [hunk 4]
```diff
@@ -27,6 +27,8 @@ contract MultisigIsm is IMultisigIsm, Ownable {
 
     // ============ Constants ============
 
+    // (required by ISM interface)
+    // solhint-disable-next-line const-name-snakecase
```
Comment (asaj) [inline solidity/contracts/isms/MultisigIsm.sol:31]: Oh, annoying. Not too late to change on our end if we want

File: solidity/contracts/isms/MultisigIsm.sol [hunk 5]
```diff
@@ -99,10 +101,12 @@ contract MultisigIsm is IMultisigIsm, Ownable {
         uint32[] calldata _domains,
         address[][] calldata _validators
     ) external onlyOwner {
-        require(_domains.length == _validators.length, "!length");
-        for (uint256 i = 0; i < _domains.length; i += 1) {
+        uint256 domainsLength = _domains.length;
```
Comment (asaj) [inline solidity/contracts/isms/MultisigIsm.sol:104]: prefix with `_` to match the rest of our style, here and everywhere else in this PR..

File: solidity/contracts/isms/MultisigIsm.sol [hunk 6]
```diff
@@ -336,7 +341,7 @@ contract MultisigIsm is IMultisigIsm, Ownable {
                 ;
                 _validatorIndex < _validatorCount &&
                     _signer != _metadata.validatorAt(_validatorIndex);
-                ++_validatorIndex
+                ++_validatorIndex // solhint-disable-next-line no-empty-blocks
```
Comment (asaj) [inline solidity/contracts/isms/MultisigIsm.sol:344]: Maybe a `while` loop would make more sense here?
```
while(_validatorIndex < _validatorCount &&  _signer != _metadata.validatorAt(_validatorIndex) {
  ++_validatorIndex;
}
```

File: solidity/test/LiquidityLayerRouter.t.sol [hunk 7]
```diff
@@ -95,7 +95,7 @@ contract LiquidityLayerRouterTest is Test {
             TypeCasts.addressToBytes32(address(destinationBridgeAdapter))
         );
         destinationBridgeAdapter.enrollRemoteRouter(
-            destinationDomain,
+            originDomain,
```
Comment (asaj) [inline solidity/test/LiquidityLayerRouter.t.sol:98]: How did a test not catch this?

Comment (yorhodes) [reply solidity/test/LiquidityLayerRouter.t.sol:98]: we weren't actually using the remote router stuff everywhere in the LL

**Pull request merged**